overlord/confdbstate/confdbmgr.go

```diff
@@ -114,52 +114,112 @@ func (m *ConfdbManager) clearOngoingTransaction(t *state.Task, _ *tomb.Tomb) err
 	return nil
 }
 
-func setOngoingTransaction(st *state.State, account, confdbName, commitTaskID string) error {
-	var commitTasks map[string]string
-	err := st.Get("confdb-commit-tasks", &commitTasks)
+type confdbTransactions struct {
+	ReadTxIDs []string `json:"read-tx-ids,omitempty"`
+	WriteTxID string   `json:"write-tx-id,omitempty"`
+}
+
+// addReadTransaction adds a read transaction for the specified confdb, if no
+// write transactions is ongoing.
+func addReadTransaction(st *state.State, account, confdbName, id string) error {
+	txs, updateTxStateFunc, err := getOngoingTxs(st, account, confdbName)
```

Comment:
Wouldn't this suffer from TOCTOU if called without the state locked?

Reply:
Given how `getOngoingTxs` access the state directly, I am guessing this should have been already locked by the called? Mentioning that the state should be locked by the caller in the doc comment should make it clear.

Reply:
Yes, these helpers will panic if called without the state locked and the closure is not meant to be long lived. I'll add a comment clarifying, thanks